### PR TITLE
added research on validation packages

### LINKED ISSUE #1: Research existing validation libraries

### PR DESCRIPTION
**Pull request by nschlemm**
fixes #1

### CODE REVIEW

**Comment by phalt** [comment]
Few, that's quite a few examples! Nice work :)

**Comment by nschlemm** [comment]
thanks! I tried sorting out those that had look unmaintained and picking representative examples for some unique or neat features like the chained validation and transformation from https://github.com/leforestier/naval#validation-and-transformation posted on slack previously

File: docs/source/research.rst [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+            [
+                Assert(
+                    (lambda d: d['password'] == d['password2']),
+                    error_message = "Passwords don't match"
+                )
+            ],
+            ['password', lambda s: s.encode('utf-8'), bcrypt.encrypt, Save],
+            ['password2', Delete],
+            ['email', Email]
+        )
```
Comment (Ian-Foote) [inline docs/source/research.rst]: This looks interesting!

File: docs/source/research.rst [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+    ...     __schema__ = { "name": Name, "email": str }
+    ...
+    >>> jdoe = User(name=Name(first="John", family="Doe"),
+    ...             email="j@doe.org")
+    ...
+
+
+`marshmallow: simplified object serialization <https://marshmallow.readthedocs.org/>`_
+--------------------------------------------------------------------------------------
+*marshmallow is an ORM/ODM/framework-agnostic library for converting complex datatypes, such as objects, to and from
```
Comment (MoritzS) [inline docs/source/research.rst]: At least github doesn't render this line correctly, maybe rst doesn't support multiline `*foo*`?

File: docs/source/research.rst [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+    ...                               UniqueUsername())
+    ...     password = SecurePassword()
+    ...     password_confirm = validators.ByteString()
+    ...     chained_validators = [validators.FieldsMatch(
+    ...         'password', 'password_confirm')]
+
+
+`Gladiator <https://github.com/laco/gladiator>`_
+------------------------------------------------
+*Gladiator is a Data Validation Framework for Python3 *
```
Comment (Ian-Foote) [inline docs/source/research.rst]: Remove the extra space before the final `*`.

**Comment by MoritzS** [comment]
I fixed the formatting errors and "pep8-ified" the code examples. Merged in 2d4a41912fb89f22edf07dbcb102c6aaac937e59!

**Pull request closed without merging**